src/niweb/apps/noclook/forms/common.py

```diff
@@ -963,3 +980,26 @@ def __init__(self, *args, **kwargs):
         self.fields['relationship_member_of'].choices = get_node_type_tuples('Contact')
 
     relationship_member_of = relationship_field('contact', True)
+
+    def clean(self):
+        if 'relationship_member_of' in self.data:
+            self.data = self.data.copy()
+            del self.data['relationship_member_of']
+
+
+class NewRoleForm(forms.ModelForm):
+    class Meta:
+        model = Role
+        fields = '__all__'
+
+class EditRoleForm(forms.ModelForm):
+    def save(self, commit=True):
+        role = super(EditRoleForm, self).save(commit)
```

Comment:
I think this is a case when we should use commit=False and then do the actual commit after the graph database operation has returned.

Reply:
Right, this would make sense once the changes in norduniclient are implemented, which will be asap.

Reply:
The code for this and the comment below has been changed.